### PR TITLE
Pytorch notebook 3.7 draft

### PR DESCRIPTION
**Pull request by GZHoffie**
Here is the first draft of pytorch notebook 3.7. There are several problems:

1.  The `pytorch` estimator `Estimator.get_validation_summary` and `Estimator.get_train_summary` are not working, both returning `None` while it should be returning a 2D-array according to the description. Currently I have to evaluate on the test set at every epoch to get the training history, which is extremely slow.
2.  To repeatedly fit the estimator for 1 epoch, I need to write the following:
     ```python
     for i in range(num_epochs):
         est.fit(data=train_loader, epochs=i+1, checkpoint_trigger=EveryEpoch())
     ```
     so that the estimator keeps training in the `for` loop. If I write `est.fit(..., epochs=1)`, then the estimator will only train for 1 epoch regardless of `num_epochs`. This might be due to the checkpoint, but I think this code is not so consistent with intuition.

### CODE REVIEW

**Comment by hkvision** [comment]
Why the file is so large (12M)?

**Comment by GZHoffie** [comment]
> Why the file is so large (12M)?

Sorry, I forgot to delete the output of a cell. Should be ok now.

**Comment by hkvision** [comment]
It is strange to still use keras dataset. Is there any options to replace this (any pytorch equivalent dataset or we write the data loading by ourselves?)

File: pytorch/3.7-predicting-house-prices.ipynb [hunk 1]
```diff
@@ -0,0 +1,656 @@
...
+    "    val_loader = data_loader_creator(val_data, val_targets, batch_size=1, shuffle=False)\n",
+    "\n",
+    "    # Build the Keras model (already compiled)\n",
+    "    model = RegressionNet()\n",
+    "    model.train()\n",
+    "    est = Estimator.from_torch(model=model, optimizer=optimizer, loss=criterion, metrics=[MAE()])\n",
+    "    # Train the model (in silent mode, verbose=0)\n",
+    "    results = []\n",
+    "    for i in range(num_epochs):\n",
+    "        est.fit(data=train_loader, epochs=i+1, checkpoint_trigger=EveryEpoch())\n",
```
Comment (hkvision) [inline pytorch/3.7-predicting-house-prices.ipynb:427]: why epochs is i+1 not 1? Is the implementation of bigdl backend continues the previous epoch?

**Comment by hkvision** [comment]
It looks good.

File: pytorch/3.7-predicting-house-prices.ipynb (hunk 1 above)
Comment (GZHoffie) [reply pytorch/3.7-predicting-house-prices.ipynb:427]: As described above, if I write 1 then the estimator will only train for 1 epoch at the first loop and won't train in the rest of the loops, which is a bit strange...

Comment (hkvision) [reply pytorch/3.7-predicting-house-prices.ipynb:427]: @qiuxin2012 We need to fix this issue. Seems previously in KerasNet we have done this, not sure can migrate to here.

**Comment by GZHoffie** [comment]
> It is strange to still use keras dataset. Is there any options to replace this (any pytorch equivalent dataset or we write the data loading by ourselves?)

I searched but there is only scikit-learn equivalent dataset. There won't be any difference since I converted the dataset into torch `DataLoader`.

**Comment by hkvision** [comment]
> > It is strange to still use keras dataset. Is there any options to replace this (any pytorch equivalent dataset or we write the data loading by ourselves?)
> 
> I searched but there is only scikit-learn equivalent dataset. There won't be any difference since I converted the dataset into torch `DataLoader`.

The main concern is it is weird for pytorch users to install and use keras utilities....

**Comment by hkvision** [comment]
@cyita Take a look at the train summary issue described above.

**Comment by cyita** [comment]
@GZHoffie Please try to call `est.set_tensorboard` before `Estimator.get_validation_summary` and `Estimator.get_train_summary`

**Comment by GZHoffie** [comment]
> @GZHoffie Please try to call `est.set_tensorboard` before `Estimator.get_validation_summary` and `Estimator.get_train_summary`

It is working, thanks a lot! I will update this notebook very soon. 

A note: I called `est.set_tensorboard` before `est.fit` to make it work.

File: pytorch/3.7-predicting-house-prices.ipynb (hunk 1 above)
Comment (qiuxin2012) [reply pytorch/3.7-predicting-house-prices.ipynb]: @hkvision   This `epoch` means the end epoch of this `fit`, not how many epoch to train. It's right behaviour.
